### ISSUE TITLE
Generated GlideOptions are missing annotations

### ISSUE THREAD

**Issue by TWiStErRob** [bumptech/glide#2774]
**Glide Version**: 4.5.0-SNAPSHOT

**Integration libraries**: none

**Device/Android Version**: none

**Issue details / Repro steps / Use case background**: Just looking around and noticed that `com.bumptech.glide.request.RequestOptions#centerCrop` has a warning on it saying "overriding methods are missing `@NonNull`". I checked where and it turns out it's from APT code. `FlickrGlideExtension` defines some of those methods, I wonder if it should warn/fail during processing that it's missing an annotation, or just add it automatically. Probably latter since all `RequestOptions` returning methods are builders and need to chain. On the other hand not having `@NonNull` in the extension class allows to return null without warning.

**Glide load line / `GlideModule` (if any) / list Adapter code (if any)**:
https://github.com/bumptech/glide/blob/v4.4.0/samples/flickr/src/main/java/com/bumptech/glide/samples/flickr/FlickrGlideExtension.java

**Comment by TWiStErRob** [comment]
See also `com.bumptech.glide.RequestManager#as`

**Comment by sjudd** [comment]
Yeah and embarrassingly the centerCrop that's in that code is for testing and shouldn't have been committed :/.

**Comment by sjudd** [comment]
I'll add @NonNull by default to any methods generated from extensions and also warn if the extension itself is missing the annotation.
